tianshou/data/batch.py

```diff
@@ -501,50 +544,97 @@ def to_torch(self, dtype: Optional[torch.dtype] = None,
             elif isinstance(v, Batch):
                 v.to_torch(dtype, device)
 
-    def cat_(self,
-             batches: Union['Batch', List[Union[dict, 'Batch']]]) -> None:
-        """Concatenate a list of (or one) :class:`~tianshou.data.Batch` objects
-        into current batch.
+    def __cat(self,
+              batches: Union['Batch', List[Union[dict, 'Batch']]],
+              lens: List[int]) -> None:
+        """
+        If x=Batch(a=Batch(a=np.random.randn(3, 4)), b=np.random.randn(3, 4)),
+        y = Batch(a=Batch(a=Batch()), b=np.random.randn(3, 4)), if we want to
+        concatenate x and y, we want to pad y.a.a with zeros. Without ``lens``
+        as a hint, when we concatenate x.a and y.a, we would not be able to
+        know how to pad y.a. So we need ``Batch.cat_`` to compute the ``lens``
+        to give ``Batch.__cat`` a hint.
```

Comment:
```python
In [27]: b=Batch(a=np.zeros([3,4]),b=np.zeros([5,4]),c={})

In [28]: c=Batch(a=np.zeros([4,4]),b=np.zeros([5,4]),c=np.zeros([1,4]))

In [29]: Batch.cat([b,c]).c.shape
Out[29]: (4, 4)
```
Please specify this undefined behavior in the documentation or unittest.

Reply:
This should go into the doc string of ``len``.